src/calibre/ebooks/epub/cfi/parse.py

```diff
@@ -57,6 +57,9 @@ def __init__(self):
     def parse_epubcfi(self, raw):
         ' Parse a full epubcfi of the form epubcfi(path [ , path , path ]) '
         null = {}, {}, {}, raw
+        if not raw:
+            return null
+
```

Comment:
This just eliminates an error message, since if raw is not valid an exception will be thrown.